lib/core/src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -1870,3 +1888,60 @@ invariant msg a predicate =
 distance :: (Ord a, Num a) => a -> a -> a
 distance a b =
     if a < b then b - a else a - b
+
+-- | Source of Stake Pool Metadata aggregation.
+--
+-- Don't export 'MkFetchSMASH' contructors, so that @fromText@ is our smart constructor.
```

Comment:
Another way you could do it, which might be shorter, is to make
```haskell
newtype SmashServer { unSmashServer :: URI } deriving (Show, Generic, Eq)
````
Use `fromText` as the smart constructor of `SmashServer`, and define `PoolMetadataSource` using `SmashServer`, without view patterns.

Reply:
I hope this doesn't mess with our json encoding.